examples/new_simulation.py

```diff
@@ -0,0 +1,431 @@
+
+# Add noise
+sens=1
+knee=1
+ylf=1
+fsky=0.1
+nell=np.zeros([len(nu),lmax+1])
+_,nell[:,2:],_=Simons_Observatory_V3_SA_noise(sens,knee,ylf,fsky,lmax+1,1)
+nell*=cl2dl[None,:]
+
+N_ells_sky = np.zeros([6, 2, 6, 2, len(ells)])
+for i,n in enumerate(nu):
+    for j in [0,1]:
+        N_ells_sky[i, j, i, j, :] = nell[i]
+
+if o.do_simulation:
+    maps_comp = np.zeros([3, 2+1, npix])
+    maps_comp[0,:,:] = hp.synfast([cl_cmb_tt, cl_cmb_ee, cl_cmb_bb, cl_cmb_te, cl_cmb_eb, cl_cmb_tb], nside, new=True)
+    maps_comp[1,:,:] = hp.synfast([cl_dust_tt, cl_dust_ee, cl_dust_bb, cl_dust_te, cl_dust_eb, cl_dust_tb], nside, new=True)
+    maps_comp[2,:,:] = hp.synfast([cl_sync_tt, cl_sync_ee, cl_sync_bb, cl_sync_te, cl_sync_eb, cl_sync_tb], nside, new=True)
+
+    if o.pysm_sim:
+            d2 = models("d2", nside) 
+            s1 = models("s1", nside) 
+            c1 = models("c1", nside)
+            
+            map_I_dust, map_Q_dust, map_U_dust = maps_comp[1,:,:]
+            map_I_sync, map_Q_sync, map_U_sync = maps_comp[2,:,:]
+            map_I_cmb,map_Q_cmb,map_U_cmb = maps_comp[0,:,:]
+
+            # Dust
+            d2[0]['A_I'] = map_I_dust
+            d2[0]['A_Q'] = map_Q_dust
+            d2[0]['A_U'] = map_U_dust
+            d2[0]['spectral_index'] = beta_dust
+            d2[0]['temp'] = temp_dust * np.ones(d2[0]['temp'].size) #need array, no const value for temp with PySM
+            # Sync
+            s1[0]['A_I'] = map_I_sync
+            s1[0]['A_Q'] = map_Q_sync
+            s1[0]['A_U'] = map_U_sync
+            s1[0]['spectral_index'] = beta_sync
+            # CMB
+            c1[0]['A_I'] = map_I_cmb
+            c1[0]['model'] = 'pre_computed' #different output maps at different seeds 
+            c1[0]['A_Q'] = map_Q_cmb
+            c1[0]['A_U'] = map_U_cmb
+            
+            sky_config = {'dust' : d2, 'synchrotron' : s1, 'cmb' : c1}
+            
+            sky = pysm.Sky(sky_config)
+
+            beams=Simons_Observatory_V3_SA_beams() 
+            instrument_config = {
+                    'nside' : nside,
+                    'frequencies' : nu, #Expected in GHz 
+                    'use_smoothing' : False,
+                    'beams' : beams, #Expected in arcmin 
+                    'add_noise' : False,
+                    'use_bandpass' : False,
+                    'channel_names' : ['LF1', 'LF2', 'MF1', 'MF2', 'UHF1', 'UHF2'],
+                    'output_units' : 'uK_RJ',
+                    'output_directory' : dirname,
+                    'output_prefix' : '/test_deltabpass_',
+            }
+    
+            sky = pysm.Sky(sky_config)
+            instrument = pysm.Instrument(instrument_config)
+            maps_signal, _ = instrument.observe(sky, write_outputs=False)
+            maps_signal = maps_signal[:,1:,:]
+            maps_signal = maps_signal/f_cmb_RJ[:,None,None]
+    else:
+            sed_comp = np.array([f_cmb, f_dust, f_sync]).T
```

Comment:
Ok, I think this is wrong. Let's think about this:
 - `np.array([f_cmb, f_dust, f_sync])` has shape `[3, n_nu, n_pix]`
 - Therefore `np.array([f_cmb, f_dust, f_sync]).T` will have shape `[n_pix, n_nu, 3]`

You want `sed_comp` to have shape `[n_nu, 3, n_pix]`, and that's not achieved by just reshaping it.
Instead, you need to do:
```
# This will swap the axes 0 and 1
sec_comp = np.transpose(np.array([f_cmb, f_dust, f_sync]), axes=[1, 0, 2])
```

And remove the `reshape` command in the next line.